src/Compilers/CSharp/Portable/Lowering/LocalRewriter/LocalRewriter_IsPatternOperator.cs

```diff
@@ -15,22 +15,16 @@ internal sealed partial class LocalRewriter
         public override BoundNode VisitIsPatternExpression(BoundIsPatternExpression node)
         {
             BoundDecisionDag decisionDag = node.GetDecisionDagForLowering(_factory.Compilation);
-            bool negated = node.IsNegated;
             BoundExpression result;
-            if (canProduceLinearSequence(decisionDag.RootNode, whenTrueLabel: node.WhenTrueLabel, whenFalseLabel: node.WhenFalseLabel))
+            if (decisionDag.RootNode is BoundLeafDecisionDagNode l && l.Label == node.WhenFalseLabel)
             {
-                // If we can build a linear test sequence `(e1 && e2 && e3)` for the dag, do so.
-                var isPatternRewriter = new IsPatternExpressionLinearLocalRewriter(node, this);
-                result = isPatternRewriter.LowerIsPatternAsLinearTestSequence(node, decisionDag, whenTrueLabel: node.WhenTrueLabel, whenFalseLabel: node.WhenFalseLabel);
-                isPatternRewriter.Free();
+                result = _factory.Literal(false);
             }
-            else if (canProduceLinearSequence(decisionDag.RootNode, whenTrueLabel: node.WhenFalseLabel, whenFalseLabel: node.WhenTrueLabel))
```

Comment:
The inverted `canProduceLinearSequence` check was preventing us to use the general rewriter which handles switch dispatch, but that is already accounted for: when IsNegated is true, the decision dag is created off of the inner pattern.

https://github.com/dotnet/roslyn/blob/85712db048f805f4fa94089df7a805fd616b7eed/src/Compilers/CSharp/Portable/Binder/Binder_Patterns.cs#L58-L60

Reply:
Also filed https://github.com/dotnet/roslyn/issues/65876